crates/runtime/src/instance.rs

```diff
@@ -604,29 +604,39 @@ impl Instance {
         // dropping a non-passive element is a no-op (not a trap).
     }
 
-    /// Do a `memory.copy` for a locally defined memory.
+    /// Do a `memory.copy`
     ///
     /// # Errors
     ///
     /// Returns a `Trap` error when the source or destination ranges are out of
     /// bounds.
-    pub(crate) fn defined_memory_copy(
+    pub(crate) fn memory_copy(
         &self,
-        memory_index: DefinedMemoryIndex,
+        dst_index: MemoryIndex,
         dst: u32,
+        src_index: MemoryIndex,
         src: u32,
         len: u32,
     ) -> Result<(), Trap> {
         // https://webassembly.github.io/reference-types/core/exec/instructions.html#exec-memory-copy
 
-        let memory = self.memory(memory_index);
+        let get_mem = |idx| {
+            if let Some(index) = self.module.defined_memory_index(idx) {
+                self.memory(index)
+            } else {
+                unsafe { *self.imported_memory(idx).from }
+            }
+        };
```

Comment:
This seems like a useful helper to factor out, in general, similar to what we have for getting tables regardless of whether they are defined/imported.

Reply:
Oh wow turns out the helper already exists and I just missed it!